bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerLegacy.java

```diff
@@ -190,100 +193,83 @@ private void readInitialInfo(Reader reader) throws RemoteControllerException {
     /**
      * Close connection to Samsung TV.
      *
-     * @throws RemoteControllerException
      */
-    public void closeConnection() throws RemoteControllerException {
+    @SuppressWarnings("null")
+    public void closeConnection() {
         try {
             if (socket != null) {
                 socket.close();
             }
         } catch (IOException e) {
-            throw new RemoteControllerException(e);
+            // ignore error
         }
     }
 
-    /**
-     * Send key code to Samsung TV.
-     *
-     * @param key Key code to send.
-     * @throws RemoteControllerException
-     */
-    @Override
-    public void sendKey(KeyCode key) throws RemoteControllerException {
-        logger.debug("Try to send command: {}", key);
+    public void sendUrl(String command) {
+        logger.warn("{}: Remote control legacy: unsupported command: {}", host, command);
+    }
 
-        if (!isConnected()) {
-            openConnection();
-        }
+    public void sendSourceApp(String command) {
+        logger.warn("{}: Remote control legacy: unsupported command: {}", host, command);
+    }
 
-        try {
-            sendKeyData(key);
-        } catch (RemoteControllerException e) {
-            logger.debug("Couldn't send command", e);
-            logger.debug("Retry one time...");
+    public void updateCurrentApp() {
+    }
 
-            closeConnection();
-            openConnection();
+    public void getArtmodeStatus(String... optionalRequests) {
+    }
 
-            sendKeyData(key);
-        }
+    public boolean closeApp() {
+        return false;
+    }
```

Comment:
What is the use if this will allways return false?

Reply:
It only returns false in this class. other services implement it with different results, and which class is used for a service depends on the TV age.

So the RemoteControllerService could be RemoteControllerLegacy or RemoteControllerWebSocket. Both classes implement all methods, but only some are supported. RemoteControllerLegacy does not support apps, RemoteControllerWebSocket does.